pygmt/datasets/map_tiles.py

```diff
@@ -0,0 +1,119 @@
+"""
+Function to load raster basemap tiles from XYZ tile providers, and load as
+:class:`xarray.DataArray`.
+"""
+
+try:
+    import contextily
+except ImportError:
+    contextily = None
+
+import numpy as np
+import xarray as xr
+
+__doctest_requires__ = {("load_map_tiles"): ["contextily"]}
+
+
+def load_map_tiles(region, source=None, lonlat=True, **kwargs):
+    """
+    Load a georeferenced raster basemap from XYZ tile providers.
+
+    The tiles that compose the map are merged and georeferenced into an
+    :class:`xarray.DataArray` image with 3 bands (RGB). Note that the returned
+    image is in a Spherical Mercator (EPSG:3857) coordinate reference system.
```

Comment:
Perhaps we should convert the returned image to longitude/latitude because it's more commonly used than the Spherical Mercator coordinate system, following https://contextily.readthedocs.io/en/latest/warping_guide.html.

Reply:
Most tiles are originally served as Web Mercator (EPSG:3857), and reprojecting to latlon (EPSG:4326) would result in distortion. I'd prefer the reprojection to be a user controlled step (e.g. using `rioxarray`'s `.rio.to_crs`), because if latlon isn't what the user wants, there would be double reprojection (EPSG:3857 -＞EPSG:4326 -＞EPSG:????) which is less accurate than single reprojection (EPSG:3857 -＞EPSG:????).

Reply:
> I'd prefer the reprojection to be a user controlled step (e.g. using `rioxarray`'s `.rio.to_crs`)

What about having an option so that users can decide on the desired projection and don't have to learn the syntax of the `rioxarray` package?

Reply:
Maybe in a separate PR?

Reply:
> Maybe in a separate PR?

Sounds good to me.